src/gtk/renderer.cpp

```diff
@@ -568,26 +570,40 @@ wxRendererGTK::GetCheckBoxSize(wxWindow* win)
         sc.Add("check");
         gtk_style_context_get(sc, GTK_STATE_FLAG_NORMAL,
             "min-width", &min_width, "min-height", &min_height, NULL);
-        GtkBorder margin;
-        gtk_style_context_get_margin(sc, GTK_STATE_FLAG_NORMAL, &margin);
-        min_width += margin.left + margin.right;
-        min_height += margin.top + margin.bottom;
+        if ( addMargins )
+        {
+            GtkBorder margin;
+            gtk_style_context_get_margin(sc, GTK_STATE_FLAG_NORMAL, &margin);
+            min_width += margin.left + margin.right;
+            min_height += margin.top + margin.bottom;
+        }
     }
     else
     {
         GValue value = G_VALUE_INIT;
         g_value_init(&value, G_TYPE_INT);
         gtk_style_context_get_style_property(sc, "indicator-size", &value);
         min_width = g_value_get_int(&value);
-        gtk_style_context_get_style_property(sc, "indicator-spacing", &value);
-        min_width += 2 * g_value_get_int(&value);
+        if ( addMargins )
+        {
+            gtk_style_context_get_style_property(sc, "indicator-spacing", &value);
+            min_width += 2 * g_value_get_int(&value);
+        }
         min_height = min_width;
         g_value_unset(&value);
     }
 
     return wxSize(min_width, min_height);
 #else // !__WXGTK3__
-    gint indicator_size, indicator_spacing;
+    gint indicator_size;
+    if ( !addMargins )
+    {
+        gtk_widget_style_get(wxGTKPrivate::GetCheckButtonWidget(),
+                            "indicator_size", &indicator_size);
```

Comment:
I'll fix this myself, but, just for the record: this is a bad bug (probably an immediate crash) as the list of arguments to this vararg function must be `NULL`-terminated.